solutions/gold/rmi-17-Hangman2.mdx

```diff
@@ -0,0 +1,154 @@
+---
+id: rmi-17-Hangman2
+source: RMI 2017 Day 1
+title: Hangman 2
+author: Justin Ji
+---
+
+<Spoiler title = "Hint">
+
+Break the problem into two cases: $n < k$ and $n \geq k$.
+
+</Spoiler>
+
+<Spoiler title = "Hint 2">
+
+Consider a $\mathcal{O}(NKmin(N, K))$ solution. What is $min(n, k)$ bounded by?
```

Comment:
> What is $min(n, k)$ bounded by?

uuuuhhhh n and k?

Reply:
i was leaning more toward them figuring out $min(n, k) \leq \sqrt{nk}$ lol, should I like make it more specific and like ask how its related to $nk$

Reply:
$min(n, k) \leq \sqrt{nk}$ seems like such a scuffed observation
like where would a square root come from

Reply:
> min(n,k)≤nk seems like such a scuffed observation like where would a square root come from

hmm idk if its common knowledge, main reason I have it stuck in my brain was cuz robezh taught that

but anyways, let $x = \sqrt{nk}$. the max $min(n, k)$ can be is equal to x, otherwise it would be the max. 

should I just give the bound in the hints lol

Reply:
@SansPapyrus683 bump

Reply:
basically LMAO

i mean, that's mostly all you need to solve the problem, along with string hashing

Reply:
it would be more accurate, but IMO it doesn't really give a good gauge of how many operations we're doing b/c only $N * K$ is bounded

Reply:
why would min not be bounded

Reply:
I mean that at least in the constraints given they only give an explicit bound on $N * K$. I can change it tho b/c I alr mentioned what $min(n, k)$ is bounded by

Reply:
hm alr
then have this hint say that "problem gives bound on n * k -> given this, what's the largest min(n, k) can be?"
it's a bit of a giveaway but i can't think of anything better
and then draw them towards casework